src/main/java/gillycarpetaddons/mixins/PhantomSpawnerMixin.java

```diff
@@ -46,12 +49,17 @@ public abstract class PhantomSpawnerMixin {
                     target = "Lnet/minecraft/entity/player/PlayerEntity;isSpectator()Z"
             )
     )
-    private boolean isSpectatorOrAtMobCap(PlayerEntity instance) {
+    private boolean isSpectatorCreativeOrAtMobCap(PlayerEntity instance) {
         boolean isSpectator = instance.isSpectator();
-        if (!GillyCarpetAddonsSettings.phantomsObeyHostileMobCap) {
-            return isSpectator;
+        if(isSpectator){
```

Comment:
Better to combine this into 1 if statement: 
`if(isSpectator || (disablePhantomSpawningForCreativePlayers && instance.isCreative()))`

For readability might even be worth introducing a private method which returns the value of `disablePhantomSpawningForCreativePlayers && instance.isCreative()`.